sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/commands.scala

```diff
@@ -113,8 +113,9 @@ case class AddFile(path: String) extends RunnableCommand {
 
   override def run(sqlContext: SQLContext): Seq[Row] = {
     val hiveContext = sqlContext.asInstanceOf[HiveContext]
+    val recursive = sqlContext.sparkContext.getConf.getBoolean("spark.input.dir.recursive", false)
```

Comment:
And by the way, I have tried:
val recursive = hiveContext.getConf("spark.input.dir.recursive", "false")
but this can only work in spark sql by execute set spark.input.dir.recursive=true before add file, and we can't set the value by --conf spark.input.dir.recursive=true. This makes it difficult for us to move some hive sql directly to SparkSQL.